### ISSUE TITLE
Can't get fonts right

### ISSUE THREAD

**Issue by jaroslavas** [marcmerlin/SmartMatrix_GFX#6]
I'm using this library on ESP32, but something's wrong with custom fonts. I've made a font using [font converter](http://oleddisplay.squix.ch/#/home), then fixed it with [font customizer](https://tchapi.github.io/Adafruit-GFX-Font-Customiser/). But when I use it, it looks different. Is there something I'm doing wrong or it's a bug in the library?

How font is supposed to look:
![Screenshot 2019-10-26 at 21 26 14](https://user-images.githubusercontent.com/1903354/67624383-85504c80-f838-11e9-8682-98cba4ff66cb.png)
 How it actually looks:
![IMG_8453](https://user-images.githubusercontent.com/1903354/67624387-8d0ff100-f838-11e9-9315-be71454d9f2b.jpg)

As you can see, something's wrong with top and bottom rows. ":" is drawn as a line, whereas it should be two dots with an empty space between them.

Here is a font file:
```
const uint8_t DSEG7_Classic_Regular_7Bitmaps[] PROGMEM = {
  0x00, 0x00, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA,
  0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA,
  0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xF0,
  0x00, 0xEA, 0xAA, 0xE0, 0x64, 0xA5, 0x29, 0x49, 0x80, 0xAA, 0xA8, 0x00,
  0xF0, 0x85, 0xE8, 0x43, 0xC0, 0xF0, 0x85, 0xE1, 0x0B, 0xC0, 0x94, 0xA5,
  0xE1, 0x08, 0x40, 0xF4, 0x21, 0xE1, 0x0B, 0xC0, 0xF4, 0x21, 0xE9, 0x4B,
  0xC0, 0xF0, 0x84, 0x21, 0x08, 0x40, 0xF4, 0xA5, 0xE9, 0x4B, 0xC0, 0xF4,
  0xA5, 0xE1, 0x0B, 0xC0, 0x88, 0x00, 0x8A, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA,
  0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xE0, 0x85,
  0xE1, 0x08, 0x00, 0x00, 0x01, 0xE1, 0x0B, 0x80, 0xF0, 0x01, 0xC0, 0x00,
  0x85, 0xE1, 0x0B, 0x80, 0xE0, 0x01, 0xE0, 0x03, 0x80, 0xE0, 0x01, 0xE0,
  0x00, 0x00, 0xE0, 0x00, 0x01, 0x0B, 0x80, 0x00, 0x01, 0xE1, 0x08, 0x00,
  0xA0, 0x00, 0x84, 0x01, 0x0B, 0x80, 0xE0, 0x01, 0xE1, 0x08, 0x00, 0x00,
  0x00, 0x00, 0x03, 0x80, 0xE0, 0x84, 0x01, 0x08, 0x00, 0xF0, 0x84, 0x00,
  0xF0, 0x85, 0xC0, 0xE0, 0x85, 0xE0, 0x00, 0x00, 0xE0, 0x85, 0xE1, 0x08,
  0x00, 0xF0, 0x00, 0x00, 0x00, 0x01, 0xE1, 0x0B, 0x80, 0x00, 0x01, 0xE0,
  0x03, 0x80, 0x10, 0xB8, 0x00, 0x84, 0x01, 0x0B, 0x80, 0x00, 0x85, 0xE1,
  0x0B, 0x80, 0x00, 0x85, 0xE1, 0x08, 0x00, 0x00, 0x85, 0xE1, 0x0B, 0x80,
  0xE0, 0x84, 0x00, 0x03, 0x80, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA,
  0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0, 0xE0,
  0x85, 0xE1, 0x08, 0x00, 0x00, 0x01, 0xE1, 0x0B, 0x80, 0xF0, 0x01, 0xC0,
  0x00, 0x85, 0xE1, 0x0B, 0x80, 0xE0, 0x01, 0xE0, 0x03, 0x80, 0xE0, 0x01,
  0xE0, 0x00, 0x00, 0xE0, 0x00, 0x01, 0x0B, 0x80, 0x00, 0x01, 0xE1, 0x08,
  0x00, 0xA0, 0x00, 0x84, 0x01, 0x0B, 0x80, 0xE0, 0x01, 0xE1, 0x08, 0x00,
  0x00, 0x00, 0x00, 0x03, 0x80, 0xE0, 0x84, 0x01, 0x08, 0x00, 0xF0, 0x84,
  0x00, 0xF0, 0x85, 0xC0, 0xE0, 0x85, 0xE0, 0x00, 0x00, 0xE0, 0x85, 0xE1,
  0x08, 0x00, 0xF0, 0x00, 0x00, 0x00, 0x01, 0xE1, 0x0B, 0x80, 0x00, 0x01,
  0xE0, 0x03, 0x80, 0x10, 0xB8, 0x00, 0x84, 0x01, 0x0B, 0x80, 0x00, 0x85,
  0xE1, 0x0B, 0x80, 0x00, 0x85, 0xE1, 0x08, 0x00, 0x00, 0x85, 0xE1, 0x0B,
  0x80, 0xE0, 0x84, 0x00, 0x03, 0x80, 0xEA, 0xAA, 0xE0, 0xEA, 0xAA, 0xE0,
  0xEA, 0xAA, 0xE0
};

const GFXglyph DSEG7_Classic_Regular_7Glyphs[] PROGMEM = {
  {     0,   1,   1,   2,    0,    0 }   // ' '
 ,{     1,   1,   1,   7,    0,    0 }   // '!'
 ,{     2,   4,   5,   4,    0,   -5 }   // '"'
 ,{     5,   4,   5,   4,    0,   -5 }   // '#'
 ,{     8,   4,   5,   4,    0,   -5 }   // '$'
 ,{    11,   4,   5,   4,    0,   -5 }   // '%'
 ,{    14,   4,   5,   4,    0,   -5 }   // '&'
 ,{    17,   4,   5,   4,    0,   -5 }   // '''
 ,{    20,   4,   5,   4,    0,   -5 }   // '('
 ,{    23,   4,   5,   4,    0,   -5 }   // ')'
 ,{    26,   4,   5,   4,    0,   -5 }   // '*'
 ,{    29,   4,   5,   4,    0,   -5 }   // '+'
 ,{    32,   4,   5,   4,    0,   -5 }   // ','
 ,{    35,   5,   1,   7,    1,   -4 }   // '-'
 ,{    36,   3,   1,   1,   -1,   -1 }   // '.'
 ,{    37,   4,   5,   4,    0,   -5 }   // '/'
 ,{    40,   5,   7,   7,    1,   -7 }   // '0'
 ,{    45,   2,   7,   7,    4,   -7 }   // '1'
 ,{    48,   5,   7,   7,    1,   -7 }   // '2'
 ,{    53,   5,   7,   7,    1,   -7 }   // '3'
 ,{    58,   5,   7,   7,    1,   -7 }   // '4'
 ,{    63,   5,   7,   7,    1,   -7 }   // '5'
 ,{    68,   5,   7,   7,    1,   -7 }   // '6'
 ,{    73,   5,   7,   7,    1,   -7 }   // '7'
 ,{    78,   5,   7,   7,    1,   -7 }   // '8'
 ,{    83,   5,   7,   7,    1,   -7 }   // '9'
 ,{    88,   2,   3,   2,    0,   -5 }   // ':'
 ,{    90,   2,   4,   4,    0,   -5 }   // ';'
 ,{    91,   4,   5,   4,    0,   -5 }   // '<'
 ,{    94,   4,   5,   4,    0,   -5 }   // '='
 ,{    97,   4,   5,   4,    0,   -5 }   // '>'
 ,{   100,   4,   5,   4,    0,   -5 }   // '?'
 ,{   103,   4,   5,   4,    0,   -5 }   // '@'
 ,{   106,   5,   7,   7,    1,   -7 }   // 'A'
 ,{   111,   5,   7,   7,    1,   -7 }   // 'B'
 ,{   116,   5,   4,   7,    1,   -4 }   // 'C'
 ,{   119,   5,   7,   7,    1,   -7 }   // 'D'
 ,{   124,   5,   7,   7,    1,   -7 }   // 'E'
 ,{   129,   5,   7,   7,    1,   -7 }   // 'F'
 ,{   134,   5,   7,   7,    1,   -7 }   // 'G'
 ,{   139,   5,   7,   7,    1,   -7 }   // 'H'
 ,{   144,   2,   3,   7,    4,   -3 }   // 'I'
 ,{   145,   5,   7,   7,    1,   -7 }   // 'J'
 ,{   150,   5,   7,   7,    1,   -7 }   // 'K'
 ,{   155,   5,   7,   7,    1,   -7 }   // 'L'
 ,{   160,   5,   7,   7,    1,   -7 }   // 'M'
 ,{   165,   5,   4,   7,    1,   -4 }   // 'N'
 ,{   168,   5,   4,   7,    1,   -4 }   // 'O'
 ,{   171,   5,   7,   7,    1,   -7 }   // 'P'
 ,{   176,   5,   7,   7,    1,   -7 }   // 'Q'
 ,{   181,   5,   4,   7,    1,   -4 }   // 'R'
 ,{   184,   5,   7,   7,    1,   -7 }   // 'S'
 ,{   189,   5,   7,   7,    1,   -7 }   // 'T'
 ,{   194,   5,   3,   7,    1,   -3 }   // 'U'
 ,{   196,   5,   7,   7,    1,   -7 }   // 'V'
 ,{   201,   5,   7,   7,    1,   -7 }   // 'W'
 ,{   206,   5,   7,   7,    1,   -7 }   // 'X'
 ,{   211,   5,   7,   7,    1,   -7 }   // 'Y'
 ,{   216,   5,   7,   7,    1,   -7 }   // 'Z'
 ,{   221,   4,   5,   4,    0,   -5 }   // '['
 ,{   224,   4,   5,   4,    0,   -5 }   // '\'
 ,{   227,   4,   5,   4,    0,   -5 }   // ']'
 ,{   230,   4,   5,   4,    0,   -5 }   // '^'
 ,{   233,   4,   5,   4,    0,   -5 }   // '_'
 ,{   236,   4,   5,   4,    0,   -5 }   // '`'
 ,{   239,   5,   7,   7,    1,   -7 }   // 'a'
 ,{   244,   5,   7,   7,    1,   -7 }   // 'b'
 ,{   249,   5,   4,   7,    1,   -4 }   // 'c'
 ,{   252,   5,   7,   7,    1,   -7 }   // 'd'
 ,{   257,   5,   7,   7,    1,   -7 }   // 'e'
 ,{   262,   5,   7,   7,    1,   -7 }   // 'f'
 ,{   267,   5,   7,   7,    1,   -7 }   // 'g'
 ,{   272,   5,   7,   7,    1,   -7 }   // 'h'
 ,{   277,   2,   3,   7,    4,   -3 }   // 'i'
 ,{   278,   5,   7,   7,    1,   -7 }   // 'j'
 ,{   283,   5,   7,   7,    1,   -7 }   // 'k'
 ,{   288,   5,   7,   7,    1,   -7 }   // 'l'
 ,{   293,   5,   7,   7,    1,   -7 }   // 'm'
 ,{   298,   5,   4,   7,    1,   -4 }   // 'n'
 ,{   301,   5,   4,   7,    1,   -4 }   // 'o'
 ,{   304,   5,   7,   7,    1,   -7 }   // 'p'
 ,{   309,   5,   7,   7,    1,   -7 }   // 'q'
 ,{   314,   5,   4,   7,    1,   -4 }   // 'r'
 ,{   317,   5,   7,   7,    1,   -7 }   // 's'
 ,{   322,   5,   7,   7,    1,   -7 }   // 't'
 ,{   327,   5,   3,   7,    1,   -3 }   // 'u'
 ,{   329,   5,   7,   7,    1,   -7 }   // 'v'
 ,{   334,   5,   7,   7,    1,   -7 }   // 'w'
 ,{   339,   5,   7,   7,    1,   -7 }   // 'x'
 ,{   344,   5,   7,   7,    1,   -7 }   // 'y'
 ,{   349,   5,   7,   7,    1,   -7 }   // 'z'
 ,{   354,   4,   5,   4,    0,   -5 }   // '{'
 ,{   357,   4,   5,   4,    0,   -5 }   // '|'
 ,{   360,   4,   5,   4,    0,   -5 }   // '}'
};

const GFXfont DSEG7_Classic_Regular_7 PROGMEM = {
(uint8_t  *)DSEG7_Classic_Regular_7Bitmaps,(GFXglyph *)DSEG7_Classic_Regular_7Glyphs,0x20, 0x7E, 8};
```

**Comment by marcmerlin** [comment]
So, the font handling comes from Adafruit::GFX, 2 libraries down.
I didn't write that code, Adafruit did.
That said, if you file a bug with them, because you have two libraries between theirs and your code (both of which I indeed did write), they aren't likely to help you either.
Honestly your best bet is to dig into the font bitmap code and check if the bitmap generated, is correct or not.
A quicker thing to try is to see whether generating from another font, gives you working results. I tried it once, and it worked for me, but I used a different font and size than you did. It's very possible that the font generator has a bug that triggered in your case.
I'll close the bug because honestly there is nothing else actionable I can do for you, but if you find useful updates, feel free to post them here for others to find. Thanks.
